### PR TITLE
Add CI for e2e

### PR DESCRIPTION
**Pull request by yizheliu-amazon**
*Description of changes:*
Set up CI for e2e. 

Basic workflow:
1. Start ES docker with AD installed
1. Check out Kibana and set up node to bootstrap
1. Start local kibana and run E2E test against it

Latest E2E run history is here: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/203

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/e2e-tests-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          plugin_version=`echo $plugin|awk -F- '{print $4}'| cut -d. -f 1-4`
+          echo $version
+          cd ..
+          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
+          then
+            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+            ## The ESRestTest Client uses http by default.
+            ## Need to disable the security plugin to call the rest api over http.
+            echo "RUN if [ -d /usr/share/elasticsearch/plugins/opendistro_security ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro_security; fi" >> Dockerfile
+            echo "RUN if [ -d /usr/share/elasticsearch/plugins/opendistro-anomaly-detection ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro-anomaly-detection; fi" >> Dockerfile
```
Comment (ohltyler) [inline .github/workflows/e2e-tests-workflow.yml:47]: Why can't we use the AD plugin from the image?

Comment (yizheliu-amazon) [reply .github/workflows/e2e-tests-workflow.yml:47]: Good catch. Will remove those unnecessary lines.

File: .github/workflows/e2e-tests-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    steps:
+      - name: Pull and Run Docker
+        run: |
+          version=1.8.0
+          echo $version
+          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
+          then
+            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+            ## The ESRestTest Client uses http by default.
+            ## Need to disable the security plugin to call the rest api over http.
```
Comment (ylwu-amzn) [inline .github/workflows/e2e-tests-workflow.yml:26]: Is it possible to test with security plugin enabled? No need to answer or change this PR if not clear answer. We can create issue and research later.

Asking this as start detector REST API will start AD job. We'd better test with security plugin enabled to know if AD job can run successfully, generate AD result and show the result correctly on Kibana.

File: .github/workflows/e2e-tests-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Run Docker Image
+        run: |
+          docker run -p 9200:9200 -d -p 9600:9600 -e "discovery.type=single-node" odfe-ad:test
+          sleep 90
+          curl -XGET http://localhost:9200/_cat/plugins
+
+      - name: Checkout Kibana
+        uses: actions/checkout@v2
+        with:
+          repository: elastic/kibana
```
Comment (ylwu-amzn) [inline .github/workflows/e2e-tests-workflow.yml:39]: Let's hold for some time for response of using `elastic/kibana` or kibana oss. Similar to this one https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/207

File: .github/workflows/e2e-tests-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Checkout Anomaly Detection Kibana plugin
+        uses: actions/checkout@v2
+        with:
+          path: kibana/plugins/anomaly-detection-kibana-plugin
+      - name: Attempt to bootstrap the plugin
+        continue-on-error: true
+        run: |
+          cd kibana/plugins/anomaly-detection-kibana-plugin
+          yarn kbn bootstrap
+      - name: Bootstrap the plugin again
```
Comment (ylwu-amzn) [inline .github/workflows/e2e-tests-workflow.yml:66]: Why bootstrap twice?

Comment (ohltyler) [reply .github/workflows/e2e-tests-workflow.yml:66]: Related to the ongoing issue (#75). When first cloning these repositories bootstrapping sometimes needs to be ran twice. Tries to reference a pulled-in dependency before it is pulled in. I've spent a decent amount of time investigating, no luck. Need to revisit

Comment (ylwu-amzn) [reply .github/workflows/e2e-tests-workflow.yml:66]: Got it. let's track in that issue (https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/issues/75)

File: .github/workflows/e2e-tests-workflow.yml (hunk 2 above)
Comment (yizheliu-amazon) [reply .github/workflows/e2e-tests-workflow.yml:26]: That's a good point. I created issue: #209

File: .github/workflows/e2e-tests-workflow.yml (hunk 4 above)
Comment (ohltyler) [reply .github/workflows/e2e-tests-workflow.yml:66]: After revisiting, I may have finally fixed this. See #75 description.

Comment (ylwu-amzn) [reply .github/workflows/e2e-tests-workflow.yml:66]: > After revisiting, I may have finally fixed this. See #75 description.

Cool, so we don't need to rerun bootstrap. That can save some time to run test workflow.

**Comment by ohltyler** [comment]
How about we change this to just run on pushes to master for now so we can merge and start using cypress?

**Comment by ylwu-amzn** [comment]
> How about we change this to just run on pushes to master for now so we can merge and start using cypress?

Totally agree. No clear reply yet. We don't need to wait for the decision. Can add PR triggered test workflow later.

**Comment by yizheliu-amazon** [comment]
> How about we change this to just run on pushes to master for now so we can merge and start using cypress?

Sure. I will use private repo and change to only run it on pushes to master.

**Comment by yizheliu-amazon** [comment]
Will verify the change after this is merged

**Pull request merged**